canonical-web-and-design/maas-ui
# Build KVM compose machine form with default interface/storage constraints.

## Done

- Built KVM compose form without the interface or storage constraint sections (so just the normal inputs)

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to a KVM details page and choose "Compose" from the action dropdown
- Check that defaults are loaded in correctly and the help text for cores and memory is correct
- Change some values and click "Compose machine"
- Check that a notification appears confirming the machine is created, and the new machine appears in the KVM's machine list.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#1961

## Screenshot
![0 0 0 0_8400_MAAS_r_kvm_185](https://user-images.githubusercontent.com/25733845/87739837-1bdbf800-c824-11ea-9e3e-5a3790e17e5c.png)
